src/components/transport_manager/src/bluetooth/bluetooth_device_scanner_posix.cc

```diff
@@ -497,5 +496,11 @@ void BluetoothDeviceScanner::BluetoothDeviceScannerDelegate::threadMain() {
   scanner_->Thread();
 }
 
+void BluetoothDeviceScanner::BluetoothDeviceScannerDelegate::exitThreadMain() {
+  scanner_->shutdown_requested_ = true;
```

Comment:
@vlantonov,
Do we guaranty that all reading in `threadmain`is blocking in less then a few second and non-blocking?
I see a risk the same behavior like in `mqueue_adapter.cc`.